db/migrate/20230613153019_add_foreign_key_constraint_to_application_id_on_oauth_access_tokens.rb

```diff
@@ -0,0 +1,14 @@
+class AddForeignKeyConstraintToApplicationIdOnOauthAccessTokens < ActiveRecord::Migration[7.0]
+  def change
+    Doorkeeper::AccessToken.all.select { |at| at.application.nil? }.each do |access_token|
+      Doorkeeper::Application.create!(
+        id: access_token.application_id,
+        name: "Unknown #{access_token.application_id}",
+        redirect_uri: "https://example.com/",
+        description: "Added in migration (previously deleted?)",
+      )
+    end
+
+    add_foreign_key :oauth_access_tokens, :oauth_applications, column: :application_id
```

Comment:
I checked this method signature as we discussed earlier: `add_foreign_key(from_table, to_table, options = {})` - I think that's what we want to do add a fk from `oauth_access_tokens` to `oauth_applications`.

Reply:
Thanks, @chrislo!